src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
```

Comment:
PullProofAI

The new code introduces a type `ControllerResponse` and a function `APIResponse.create` for generating controller responses. Here are some suggestions for improvement:

- Consider using interfaces instead of types for better compatibility with TypeScript features.
- Ensure that the `isSuccess` property is a boolean to clearly indicate success or failure.
- Validate the input parameters to `APIResponse.create` function to handle edge cases gracefully.
- Instead of using `any` type, try to define specific types for `data` and `error` properties for better type safety.

Overall, the code looks good but can be enhanced for better type safety and error handling.

```typescript
/* eslint-disable @typescript-eslint/no-explicit-any */
export interface ControllerResponse {
  isSuccess: boolean;
  statusCode: number;
  message: string;
  data?: unknown;
  error?: Error;
}

export const APIResponse = {
  create: (
    isSuccess: boolean,
    statusCode: number,
    message?: string,
    data?: unknown,
    error?: Error,
  ): ControllerResponse => {
    const response: ControllerResponse = { isSuccess, statusCode, message };

    if (error) {
      response.error = error;
    } else if (!error && data) {
      response.data = data;
    } else {
      response.data = {};
    }

    return response;
  },
};
```

Feel free to incorporate these suggestions to improve the code quality and maintainability.


<!-- This is an auto-generated comment by PullProofAI -->